pandas/io/excel/_base.py

```diff
@@ -1106,7 +1099,6 @@ def __init__(
                     FutureWarning,
                     stacklevel=stacklevel,
                 )
-        assert engine in self._engines, f"Engine {engine} not recognized"
```

Comment:
Why remove?

Reply:
Engine should be validate by https://github.com/pandas-dev/pandas/pull/38710/files#diff-63200ddb7f5656b8ee868a28d9cb7720ffe50689b0e3fb0b4e15cc5c0ae80dd7R1035 if passed explicitly
and by set_option if set there(I will add this).